src/main/java/net/neoforged/neoforge/common/extensions/IPlayerExtension.java

```diff
@@ -81,4 +82,11 @@ default boolean mayFly() {
         // TODO 1.20.5: consider forcing mods to use the attribute
         return self().getAbilities().mayfly || self().getAttributeValue(NeoForgeMod.CREATIVE_FLIGHT) > 0;
     }
+
+    /**
+     * {@return whether this player is a fake player, such as {@link FakePlayer}}
+     */
+    default boolean isFakePlayer() {
```

Comment:
Thought to consider: would it be worth putting this on `LivingEntity`? Occasionally there is an interaction where you wish to check if its a fake player but are fine with a non-player living entity. With this method, you'd still have to do something like `living instanceof Player player && player.isFakePlayer()`.

The idea could be extended to other "fake entities" though I cannot think of a usecase of such a concept. If that were a useful thing then a name like `isFakeEntity` would be better.

Reply:
fake entities are usually real ones that just haven't been added to the world. Unlike with the player object, it's easy to do that with other entities. But yes, I can see pulling up "isFake()" to Entity making sense even if no one ever made a fake non-player entity.

Reply:
I'm not really a fan of pulling something so closely tied to players up into `LivingEntity`

Reply:
Agreed if you need to know if it is a fake player, you should check for a player first. Pattern instanceof is your friend here.